src/main/resources/avro/sequenceAnnotationmethods.avdl

```diff
@@ -0,0 +1,126 @@
+@namespace("org.ga4gh.methods")
+
+protocol SequenceAnnotationMethods {
+
+  import idl "common.avdl";
+  import idl "methods.avdl";
+  import idl "sequenceAnnotations.avdl";
+
+  // FIXME: wiggle methods not defined.
+
+  /******************  /features/search  *********************/
+
+  /**
+  This request maps to the body of `POST /features/search` as JSON.
+  */
+  record SearchFeaturesRequest {
+    /**
+    The annotation set to search within. Either `featureSetId` or
+    `parentId` must be non-empty.
+    */
+    union { null, string } featureSetId;
+
+    /**
+    Restricts the search to direct children of the given parent `feature`
+    ID. Either `featureSetId` or `parentId` must be non-empty.
+    */
+    union { null, string } parentId;
+
+    /**
+    If specified, this query matches only annotations which overlap this path.
+    */
+    /* TODO: path not used yet, as graph code is not working in server.
+       Fields below are copied directly from variants.avdl for now.
+    union { null, org.ga4gh.models.Path } path = null; */
+
+    /**
+    Only return features with on the reference with this name.  One of this
+    field or `referenceId` is required.  (case-sensitive, exact match)
+    */
+    union { null, string } referenceName = null;
+
+    /**
+    Only return feature on the reference with this ID. One of this field or
+    `referenceName` is required.
+    */
+    union { null, string } referenceId = null;
+
+    /**
+    Required. The beginning of the window (0-based, inclusive) for which
+    overlapping features should be returned.  Genomic positions are
+    non-negative integers less than reference length.  Requests spanning the
+    join of circular genomes are represented as two requests one on each side
+    of the join (position 0).
+    */
+    long start;
+
+    /**
+    Required. The end of the window (0-based, exclusive) for which overlapping
+    features should be returned.
+    */
+    long end;
+
+    // TODO: Fix this field. Clarify semantics around how OntologyTerm
+    // matching works, or change the Ontology term field on the feature.
+    /**
+    If specified, this query matches only annotations which match one of the
+    provided feature types.
+    */
+    array<org.ga4gh.models.OntologyTerm> features = [];
```

Comment:
Since an empty array returns all otherwise matching features, it would follow that a non-empty array of `OntologyTerm`s would return an _intersection_ of the results on each (logical `and`). A logical `or` can always be obtained by simply performing several searches, then taking the union of their results.